src/components/PopupWithForm.js

```diff
@@ -5,24 +5,23 @@ export default class PopupWithForm extends Popup {
     super(popupSelector);
     this._handleFormSubmit = handleFormSubmit;
     this._popupForm = this._popupElement.querySelector(".modal__form");
-    this._inputs = this._popupElement.querySelectorAll(".modal__input");
-    this._submitButton = this._popupElement.querySelector(
-      ".modal__submit-button"
+    this._inputs = Array.from(
+      this._popupForm.querySelectorAll(".modal__input"),
     );
+    this._submitButton = this._popupElement.querySelector(".modal__button");
+    this._submitButtonText = this._submitButton.textContent;
   }
 
   close() {
-    this._popupForm.reset();
     super.close();
+    this._popupForm.reset();
   }
 
   _getInputValues() {
-    const _inputs = Array.from(
-      this.popupForm.querySelectorAll(".modal__input")
-    );
-    this._inputValues = {};
-      _inputs.forEach((input) => {
-      this._inputs[input.name] = input.value;
+    const _inputs = this._inputs;

```

Comment:
Needs correction: please remove this line, you don't need to duplicate value in the local constant